### ISSUE TITLE
Compatibility with Keras Model?

### ISSUE THREAD

**Issue by ahundt** [zsdonghao/tensorlayer#99]
Can I use a keras model to generate my TF layers and then train with tensorlayer, or does tensorlayer define attributes that would be incompatible?

**Comment by zsdonghao** [comment]
@ahundt  good idea. As TL is now compatible with TF-Slim, we can also make it compatible with keras ~ let me think .

**Comment by zsdonghao** [comment]
Hi, TensorLayer is compatibility with Keras in the master verison now!

here is a small example: https://github.com/zsdonghao/tensorlayer/blob/master/example/tutorial_keras.py

**Comment by ahundt** [comment]
Amazing :-) that's the fastest I've ever seen a feature implemented, you deserve a medal 🏅! I'm definitely going to install TL now!

Perhaps TL has its own equivalent some of the [training callbacks](https://github.com/fchollet/keras/blob/keras-2/keras/callbacks.py) like ReduceLROnPlateau etc?

**Comment by zsdonghao** [comment]
@ahundt Thanks. `ReduceLROnPlateau` is for changing the learning rate right?

TL does not has the opt equivalent to this, because to control the training deeply we hope people to use TF APIs.

I suggest you to define a optimizer like that:
```
with tf.variable_scope('learning_rate'):
        lr = tf.Variable(0.0, trainable=False)
optimizer = tf.train.GradientDescentOptimizer(lr).minimize(cost)
```
then you can update your learning rate during training like that: 
```
sess.run(tf.assign(lr, your_new_lr)
```
The PTB example use this method: https://github.com/zsdonghao/tensorlayer/blob/master/example/tutorial_ptb_lstm_state_is_tuple.py

**Comment by ahundt** [comment]
Awesome, thanks a lot! Since the original question & extra API questions are both resolved very well I'll close the issue.
